src/runtime/server/content-index.ts

```diff
@@ -39,7 +39,13 @@ export async function getIndexedContentsList<T = ParsedContent> (event: H3Event,
       .filter(key => (path as any).test ? (path as any).test(key) : key === String(path))
       .flatMap(key => index[key])
 
-    const contents = await Promise.all(keys.map(key => getContent(event, key)))
+    const keyChunks = [...chunksFromArray(keys, 10)]
+
+    const contents = []
+    for (const chunk of keyChunks) {
```

Comment:
```suggestion
    for await (const chunk of keyChunks) {
```

Chunks should process in sequence, without await chunks will run in paralel